### DISCUSSION TITLE
Realtime Eventsource error on Node-Red

### DISCUSSION THREAD

**Discussion by mariodelriojimenezdam** [pocketbase/pocketbase#4338] [Q&A]
Hi guys,

I'm using PocketBase with Node-Red and everything works fine, but when I try to use realtime api, it throws an error with EventSource.

I have added the dependency on package.json and settings.js, and set it like value of global.EventSource, but this doesn't solve the problem. The value of eventsource inside the PocketBase object is null.

Does anyone know how to solve the problem? Thanks in advance.

**Comment by mariodelriojimenezdam** [comment]
I have investigated how global variables are accessed in node red, and they differ from how they are accessed in node.js.
To access global variables in Node-Red you must execute the **get** method of the global object.

Could the **EventSource** class be added as a **parameter** to the Pocketbase object instead of obtaining it from the global context?

**Reply by ganigeorgiev** [reply]
> Could the EventSource class be added as a parameter to the Pocketbase object instead of obtaining it from the global context?

No. There are plans to remove `EventSource` entirely and replace it with a custom `fetch` stream but this approach has issues with React Native and it is put on hold for now.

I have no experience with Node-RED and I don't understand what is the issue here. If native/global objects can't be accessed how do you send regular requests, or do you specify custom `fetch` function as part of the SendOptions?

**Reply by mariodelriojimenezdam** [reply]
In Node.js you can access global variables via global.variableName, but in Node-Red you access global variables via global.get('variableName'), so when PocketBase object try to get EventSource class in Node-Red it obtains null.

**Reply by ganigeorgiev** [reply]
That's ok but when sending regualar requests we also use whatever global/native `fetch` implementation is available so if your script doesn't have access to the global `this` scope then the JS SDK wouldn't work at all which contradicts with your initial statement that "_everything works fine_".

`global` in this case could be just a custom object/variable available to your script but there should be a way to reach to the global `this` object. You can try with [`globalThis`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) (_or maybe try to attach to the top level `this`_).

**Reply by ganigeorgiev** [reply]
Or another option could be to register `EventSource` as constructor wrapper function, something like:

```js
function EventSource() {
   return yourPolyfillConstructor.apply(this, args)
}

const sse = new EventSource(url) // should return the polyfill
```
